frontend/public/components/impersonate-notifier.jsx

```diff
@@ -1,24 +1,33 @@
-import * as _ from 'lodash-es';
 import * as React from 'react';
 import { connect } from 'react-redux';
+import { useTranslation, Trans } from 'react-i18next';
 
 import * as UIActions from '../actions/ui';
+import { modelFor } from '../module/k8s';
 
 export const ImpersonateNotifier = connect(({ UI }) => ({ impersonate: UI.get('impersonate') }), {
   stopImpersonate: UIActions.stopImpersonate,
 })(({ stopImpersonate, impersonate }) => {
+  const { t } = useTranslation();
   if (!impersonate) {
     return null;
   }
   return (
     <div className="co-global-notification">
       <div className="co-global-notification__content">
         <p className="co-global-notification__text">
-          <span className="text-uppercase co-global-notification__impersonate-kind">{`Impersonating ${impersonate.kind}`}</span>{' '}
-          You are impersonating{' '}
-          <span className="co-global-notification__impersonate-name">{impersonate.name}</span>. You
-          are viewing all resources and roles this {_.toLower(impersonate.kind)} can access.{' '}
-          <a onClick={stopImpersonate}>Stop Impersonation</a>
+          <span className="text-uppercase co-global-notification__impersonate-kind">
+            {t('public~Impersonating {{kind}}', {
+              kind: t(modelFor(impersonate.kind)?.labelKey) || impersonate.kind,
+            })}
+          </span>{' '}
+          <Trans t={t} ns="public">
+            You are impersonating{' '}
+            <span className="co-global-notification__impersonate-name">{impersonate.name}</span>.
+            You are viewing all resources and roles this {impersonate.kind.toLowerCase()} can
+            access.{' '}
+          </Trans>
+          <a onClick={stopImpersonate}>{t('public~Stop impersonation')}</a>
```

Comment:
Not new, but this is an accessibility issue since there is no `href`. We should change this to a button styled like a link.

Reply:
good thinking, updated this to a button with a link variant.

Reply:
I had to update the css as well.